Add "Announcing Clippy feature freeze post (#1613)
rust-lang/blog.rust-lang.org: content/the-incoming-summer-of-clippy.md

Author: I'm not sure, this serves as a half announcement to overall the Rust community (they won't be getting new features for a while) and also to the contributors, as they now have distinct goals to aim to. (That tracking issue)

Reviewer: I don't think it will be really noticable to non-contributors, if there aren't any new lints for 2 releases.

The internal-rust blog is for the contributors.

Reviewer 2: I agree that this should be an "inside-rust" post. I think most users don't expect new clippy lints to be released every six weeks.

+++
path = "9999/12/31/the-incoming-summer-of-clippy"
title = "The incoming Summer of Clippy"
authors = ["blyxyas"]

[extra]
team = "the Clippy team"
team_url = "https://www.rust-lang.org/governance/teams/dev-tools#team-clippy"
+++

> We are currently working on a feature freeze for the Clippy project. This
> period would be 12 weeks long, starting on the introduction of 1.89 to beta
> on the 26 of June, and ending on the 18th of September.

Reviewer 3: So, after reading the whole blog post I understand that from end of June until mid-September Clippy won't accept new feature requests but only fixes to existing lints - is that correct?

I think a TL;DR; of this kind could help, since the title is nice :sunglasses:  but doesn't explicit what is really being communicated.

How about this slightly reworded version:
```suggestion
The Clippy project will be on feature-freeze for 12 weeks, starting from Rust 1.89.0 beta (June 26th 2025) to September 18th 2025 (Rust 1.89.0 stable release). During this time no new features will be accepted, only bug fixes.
```

Author: I agree that this might be clearer to read, maybe not an explicit TL;DR because it's less than 500 words.

This *summer of Clippy* comes from a lack of the necessary capacity needed
to maintain all the current lints (about 750 of them 😱) and still add new ones.
We need to care for the Clippy project the same way that Clippy cares about our
code, and note that every single one of them needs maintaining and accounting
for current and future changes in the compiler as a whole.

As a team we’ll focus on making contributing the best it can be, and mark the

> issues that could be great as a starting point to get familiar with the project.

Reviewer: I would move this to the `Can I help?` section.

Author: Imo, having this section in the introductory section gives to the reader some structure. Like:

What's happening -> Why it's happening -> What we'll be doing as a team. The other two sections are separate as they're meant to be read as a more involved user/contributor.

If you have a lint which you want to be implemented, you can still open an
issue for it, but no one will implement it (also, lint-adding pull requests
opened after the starting date will be blocked for the duration of the period).

Note that pull requests opened before the starting date can still be reviewed
and merged, but unless they are already in a very advanced state they will not
be prioritized over bugfixes.

## Can I help?

As always, any help on the project is very appreciated. Apart from opening bug
reports, you can follow [our contributing guidelines] and check [our suggested
good first issues] for opening pull requests . Everything from fixing diagnostic
issues and improving documentation, to fixing application-wide crashes
really helps Clippy get into the excellent state that our users deserve.

If you need direct help when fixing a bug, don't hesitate to open a thread on
Zulip, or to open the PR with the questions you have in order to start the
feedback stream.

## Conclusion

Thanks a lot for all these years of supporting Clippy, we hope that via this
little feature hiatus we can achieve the excellence that the Rust user base

> deserves (if that's even possible).

Reviewer: ```suggestion
deserves.
```

Don't sell us short; Clippy is great! :tada:

Author: It was more about how to Rust community is so good that they deserve something that nobody can deliver (but I suppose that by using Rust we are able to deliver on that promise?).

Reviewer 4: We can have nice things.

[our contributing guidelines]: https://github.com/rust-lang/rust-clippy/blob/master/CONTRIBUTING.md
[our suggested good first issues]: <!-- ADD HERE OUR TRACKING ISSUE -->

> 
> 

Reviewer: NIT: extra new lines.

```suggestion
```

+++

We are currently working on a feature freeze for the Clippy project. This
period would be 12 weeks long, starting on the introduction of 1.89 to beta
on the 26 of June, and ending on the 18th of September.

This *summer of Clippy* comes from a lack of the necessary capacity needed
to maintain all the current lints (about 750 of them 😱) and still add new ones.
We need to care for the Clippy project the same way that Clippy cares about our
code, and note that every single one of them needs maintaining and accounting
for current and future changes in the compiler as a whole.

As a team we’ll focus on making contributing the best it can be, we’ll try to
bisect all the issues possible and mark the issues that could be great as a
starting point to get familiar with the project.

## What can I expect as a user?

As a user you can expect to have more accurate lints, with less false negatives
and more edge cases covered. A bump into Clippy’s toolchain would not involve
the same risk for false positives as it has been until now. We hope that by
upping the standard for lints all around the project we can allow users to
expect even more from the already great linter.

> If you have a lint which you want to be implemented, you can still open and

Reviewer 4: ```suggestion
If you have a lint which you want to be implemented, you can still open an
```
Typo

issue for it, but no one will implement it (also, lint-adding pull requests

> opened after the starting date will be blocked for the duration of the period)

Reviewer 4: Missing sentence period.

Note that pull requests opened before the starting date can still be reviewed
and merged, but unless they are already in a very advanced state they will not

> be priorized over bugfixes.

Reviewer 4: ```suggestion
be prioritized over bugfixes.
```
